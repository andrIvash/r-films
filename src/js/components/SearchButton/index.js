// @flow
import React from 'react';

type Props = {
 handleSearch: (ev: SynteticInputIvent) => void,
}

const SearchButton = (props: Props) => (
  <button
    className='search-button btn btn-lg btn-primary'
    onClick={props.handleSearch}
    type='button'
    > Search
  </button>
);

export default SearchButton;
