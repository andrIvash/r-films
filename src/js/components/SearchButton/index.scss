.search-button {
  width: rem(100px);
  height: rem(40px);
  font-family: $sign-font;
  font-size: rem(14px);
  margin: 0 rem(5px);
  text-transform: uppercase;
}

.search-button {
  background-color: $color-logo;
  border-color: $color-logo;
  &:hover {
    background-color: $color-logo-dark;
    border-color: $color-logo-dark;
  }
  &:active:not(:disabled):not(.disabled) {
    background-color: $color-logo-dark;
  }
}

.header__top {
  .search-button {
    width: rem(70px);
    height: rem(30px);
    font-size: rem(9px);
    background-color: $color-layout;
    color: $color-logo;
    &:hover {
      color: $color-logo-dark;
      background-color: $color-layout-dark;
    }
    &:active:not(:disabled) {
      background-color: $color-logo;
    }

  }
}