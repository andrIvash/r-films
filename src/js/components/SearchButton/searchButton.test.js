import React from 'react';
import SearchButton from './index';

let props = {};

describe('SearchButton', () => {
  beforeEach(() => {
    props = {
      handleSearch: jest.fn(),
    };
  });
  it('should be defined', () => {
    expect(SearchButton).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = render(
      <SearchButton {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });
  it(`should calls handleSearch
    when button is clicked`, () => {
    const wrapper = shallow(
      <SearchButton {...props} /> );
    wrapper.simulate('click');
    expect(props.handleSearch).toHaveBeenCalledTimes(1);
  });
});
