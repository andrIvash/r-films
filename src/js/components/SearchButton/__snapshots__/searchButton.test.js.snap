// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render correctly 1`] = `
<button
  class="search-button btn btn-lg btn-primary"
  type="button"
>
   Search
</button>
`;
